src/runner/reporter-stream-controller.ts

```diff
@@ -0,0 +1,50 @@
+import { EventEmitter } from 'events';
+import { Writable } from 'stream';
+import { GeneralError } from '../errors/runtime';
+import { RUNTIME_ERRORS } from '../errors/types';
+
+interface Plugin {
+    name: string;
+    streamController: ReporterStreamController;
+}
+
+interface PluginInfo {
+    stream: Writable;
+    plugin: Plugin;
+}
+
+class ReporterStreamController {
+    public multipleStreamError: GeneralError | null;
+    private _pluginInfos: PluginInfo[];
+    private _task: EventEmitter;
+
+    public constructor (task: EventEmitter, reporters: any[]) {
+        this._pluginInfos = [];
+        this._task = task;
+        this.multipleStreamError = null;
+
+        reporters.forEach(({ plugin }) => {
+            plugin.streamController = this;
+        });
+    }
+
+    public ensureUniqueStream (stream: Writable, plugin: Plugin): boolean {
+        const pluginInfo = this._pluginInfos.find(info => info.stream === stream);
+
+        if (!pluginInfo)
+            this._pluginInfos.push({ stream: stream, plugin: plugin });
+        else if (pluginInfo.plugin !== plugin) {
+            const message = [pluginInfo.plugin.name, plugin.name].join(', ');
+
+            this.multipleStreamError = new GeneralError(RUNTIME_ERRORS.multipleSameStreamReporters, message);
+
+            this._task.emit('done');
```

Comment:
Can't we use `task.cancel` here?

Reply:
Task has `abort` method, but not `cancel`. The `abort` method does not dispose assets and does not resolve promises inside of reporters.

Reply:
Ok then